### ISSUE TITLE
Unexpected result when parsing imports

### ISSUE THREAD

**Issue by tlansec** [plyara/plyara#116]
Hello,

I was recently troubleshooting why some rules were being prepended with imports and found that when plyara.parse_string() takes a string input of rules it assigns the imports to all rules in the file. 

Here's some example code to illustrate:

```
import "pe"

rule some_pe_rule {
    condition:
        false
}

import "elf"

rule some_elf_rule {
    condition:
        false 
}
```

Example code:

```
import plyara

p = plyara.Plyara()
with open('example.yar', 'r') as infile:
    data = infile.read()

print(p.parse_string(data))
```

The result is that plyara shows both rules as requiring both imports when infact there are two rules each of which required one module (imagine the condition was something meaningful)

```
[{'rule_name': 'some_pe_rule', 'start_line': 3, 'stop_line': 6, 'raw_condition': 'condition:\n        false\n', 'condition_terms': ['false'], 'imports': ['pe', 'elf']}, {'rule_name': 'some_elf_rule', 'start_line': 10, 'stop_line': 13, 'raw_condition': 'condition:\n        false \n', 'condition_terms': ['false'], 'imports': ['pe', 'elf']}]
```

The only possible fix I can think of is for plyara to look at the condition for **import.blah** style conditions in order to determine if imports are used in a rule. 

Cheers,
Tom

**Comment by utkonos** [comment]
@tlansec Thanks for the report. We have had a lengthy discussion about this topic in #53 

I am amenable to an additional non-default option that removes imports from rules that don't use it, but the behavior you're reporting is expected and by design.

**Comment by tlansec** [comment]
It might be the design, but I expect I will not be the last user who is surprised by this behaviour - a common use case for plyara might be:

```
import plyara 
# whatever the rebuild_rule import is
with open('foo.yar, 'r') as infile:
    data = infile.read()

p = plyara.Plyara()
parsed = p.parse_string(data)
accepted_rules = []
for rule in parsed:
     # do some check, and if it passes.
     accepted_rules.append(rebuild_rule(rule))
````

Thus ending up with a large # of unnecessary imports. I appreciate that there isn't an elegant solution to this.

If any future readers come across this thread and the library is still working as described in this issue and #53, I found the best solution is to stop using rebuild_rule if you are not modifying the contents of the rule - instead use the start_line and stop_line to build the rule from the original string again.

Cheers,
Tom

**Comment by utkonos** [comment]
@tlansec I added a new parameter `import_effects` which defaults to off so that any user who wants to see imports in rules affected by said import can do. The default behavior is exactly as you asked for here. The PR was provided by @cccs-rs. I made a minor bug fix to that change to make sure that there are no stray empty lists in the output. I'm working on unit tests to account for this change. Also, this will be a breaking change. I'll be adding notes about it in documentation and the README.

Thanks for the suggestion and sorry for the delay in getting this all done.
